src/csharp/Microsoft.Spark.Worker/Command/SqlCommandExecutor.cs

```diff
@@ -737,6 +738,25 @@ internal ArrowOrDataFrameGroupedMapCommandExecutor(Version version)
             return ExecuteArrowGroupedMapCommand(inputStream, outputStream, commands);
         }
 
+        private RecordBatch WrapArrowRecordBatchColumnsInAStruct(RecordBatch batch)
+        {
+            if (SparkEnvironment.SparkVersion >= new Version(Versions.V3_0_0))
+            {
+                ArrowBuffer.BitmapBuilder validityBitmapBuilder = new ArrowBuffer.BitmapBuilder();
+                for (int i = 0; i < batch.Length; i++)
+                {
+                    validityBitmapBuilder.Append(true);
```

Comment:
If all rows are non-null, this bitmap buffer can be empty - `ArrowBuffer.Empty`. That saves the allocations and data.